### PR TITLE
fix: [FFM-8019] Fix for error in deleting prereqs

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

Previously this error was occurring:

https://github.com/harness/harness-core-ui/assets/103498167/886a5377-358e-4d80-a08e-4d5dcd4efeec

This was because we weren't handling the final call in the delete process correctly. Instructions weren't being retrieved in time and returned as undefined. Now we do not get this error:

https://github.com/harness/harness-core-ui/assets/103498167/9491334b-3441-4463-baee-d4e732fa4e4a

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by crystal-fong — APPROVED** [review]

File: src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx [hunk 1]
```diff
@@ -124,7 +124,9 @@ export const FlagPrerequisites: React.FC<FlagPrerequisitesProps> = props => {
           .catch(err => {
             showError(get(err, 'data.message', err?.message), undefined, 'cf.patch.prereq.error')
           })
-          .finally(patch.feature.reset)
+          .finally(() => {
+            patch.feature.reset()
+          })
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx:129]: This _should_ be the same thing, but it depends on how the `patch.feature` object was written. I suppose the way it was it was losing the scope of what `this` was. Don't ya just love OOP practices and the complexities they bring in JS?

Comment (laura-mannus) [reply src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx:129]: :mild-panic: yes

**Review by knagurski — APPROVED** [review]
Does the job 👍🏻

**Comment by knagurski** [comment]
Force merged due to low coverage in the area. Increasing coverage is out of the scope of this ticket and will be addressed when migrated to FF-UI.

**Pull request merged**